### DISCUSSION TITLE
How can I getting the previous month's from current date in query?

### DISCUSSION THREAD

**Discussion by positivedeveloper36** [adonisjs/core#3825] [Help]
My code like this :

```
export default class UsersService {
  public static index(params: Record<string, any>) {
    const results = User.query()
      .select('*')
      .whereRaw("created_at > CURRENT_DATE - INTERVAL '? months'", [Env.get('MONTHS'])
      .orderBy('created_at', 'desc')
      .toQuery()
    return results
  }
}
```

The result of query like this : 

```select * from "users" where created_at > CURRENT_DATE - INTERVAL ''3' months' order by "created_at" desc```

There is a problem with the quotes. How can I remove the quotes to be like this :

`select * from "users" where created_at > CURRENT_DATE - INTERVAL '3 months' order by "created_at" desc`

Please help. Thank you

**Comment by Barbapapazes** [comment]
Hello,

First, there is an issue but could only be related to this code
```diff
- [Env.get('MONTHS'])
+ [Env.get('MONTHS')])
```

Maybe you can try to write all in binding
```ts
.whereRaw("created_at > CURRENT_DATE - INTERVAL ?", [`${Env.get('MONTHS')} months`])
```

**Comment by positivedeveloper36** [comment]
I'm sorry, I typed it wrong, but that also doesn't work
I try your binding but there exist error like this : `...created_at > CURRENT_DATE - INTERVAL $1 order by "created_at" desc - syntax error at or near "$1"`

Apparently it can't use bindinng
